### PR TITLE
feat(css): add additional CSS properties and allow dash-case index

### PR DESCRIPTION
**Pull request by CaerusKaru**
* Add newer CSS properties like Flex and Grid (and Webkit) prefixes
  to CSSStyleDeclaration
* Allow for string indexing with dash-case properties to the style
  object

### CODE REVIEW

**Comment by CaerusKaru** [comment]
@cscott

**Comment by cscott** [comment]
Please clean up this patch.  There are far too many unrelated style changes, and it now conflicts with HEAD as well.  Your patch should only change the things you need to change, not eg change quotation marks in comments.

**Comment by CaerusKaru** [comment]
@cscott Cleaned up the patch and rebased. Should be good to go 👍

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 node_modules
 npm-debug.log
 v8.log
+.idea
```
Comment (cscott) [inline .gitignore]: This doesn't belong, but it's not worth punting the whole patch over.

**Comment by cscott** [comment]
I verified that dashed indexes work in the web console in Firefox (at least), like `el.style['z-index']`.

**Pull request merged**